stdlib/public/Differentiation/ArrayDifferentiation.swift

```diff
@@ -21,27 +21,29 @@ extension Array where Element: Differentiable {
   /// multiplied with itself `count` times.
   @frozen
   public struct DifferentiableView {
+    @usableFromInline
```

Comment:
Beware, adding `@usableFromInline` introduces a newly exported symbol without availability. It would not be safe to make this change in a module that's expected to have stable ABI. Newly built code after this change may not be binary compatible with Differentiation in earlier Swift releases.

(However, as I understand it, while Differentation is being build with library evolution enabled, it is not distributed as such on any platform where Swift is ABI stable. It is also unclear if the problematic stored property accessor exports are ever actually called in practice for a `@frozen` structure like this.)

Reply:
Thanks for the review @lorentey !
Just to double check, this is considered unsafe because the struct was already decorated with @frozen? Or is using @usableFromInline without public availability generally ABI unstable? I might not fully grasp the subtlety here but would love to understand better to keep these in mind for future changes. 
But indeed Differentiation is not yet distributed to a platform that is ABI stable, so it's not an issue yet!

Reply:
@rxwei in your opinion, are we good to go here? As @lorentey points out we break ABI here but that's currently not an issue yet right? I'll add some comments to the main PR message regarding ABI stability.

As an additional question, just for my understanding. Is the following change binary incompatible? And if so why? It seems to me that we're only adding information to the interface not changing the binary layout of the struct. But I also don't have a lot of experience here so would like to understand the details if possible!
```
@frozen 
struct Thing {
    var storage: Float
}
```
-> 
```
@frozen 
struct Thing {
    @usableFromInline
    var storage: Float
}
```

Also tagging @asl